### ISSUE TITLE
FancyReporter is default on windows but don't work

### ISSUE THREAD

**Issue by eventh** [dag/attest#124]
If pygments is installed FancyReporter is the default Reporter on Windows, but the windows 'terminal' don't support the output. Maybe add a test in auto_reporter() for sys.platform == 'win32' ?

Tested with python 3.2.2 on windows 7 32bit with Attest installed from up to date git repository.

Gives the following output with FancyReporter, but looks good on PlainReporter:

```
←[02m───────────────────────────────────────────────────────────────────────────
─────←[39;49;00m
Traceback (most recent call last):←[39;49;00m
  File ←[39;49;00m←[36m"C:\Program Files\Python32\lib\contextlib.py"←[39;49;00m,
 line ←[39;49;00m←[34m66←[39;49;00m, in ←[39;49;00m__exit__←[39;49;00m
    ←[39;49;00m←[36mself←[39;49;00m.←[39;49;00mgen←[39;49;00m.←[39;49;00mthrow←[
39;49;00m(←[39;49;00m←[36mtype←[39;49;00m,←[39;49;00m ←[39;49;00mvalue←[39;49;00
m,←[39;49;00m ←[39;49;00mtraceback←[39;49;00m)←[39;49;00m
  File ←[39;49;00m←[36m"tests\test_cparser.py"←[39;49;00m, line ←[39;49;00m←[34m
54←[39;49;00m, in ←[39;49;00mbasic_types←[39;49;00m
    ←[39;49;00m←[34massert←[39;49;00m ←[39;49;00ma←[39;49;00m.←[39;49;00msize←[3
9;49;00m ←[39;49;00m==←[39;49;00m ←[39;49;00m←[34m3←[39;49;00m ←[39;49;00m←[35ma
nd←[39;49;00m ←[39;49;00mb←[39;49;00m.←[39;49;00msize←[39;49;00m ←[39;49;00m==←[
39;49;00m ←[39;49;00m←[34m4←[39;49;00m ←[39;49;00m←[35mand←[39;49;00m ←[39;49;00
mc←[39;49;00m.←[39;49;00msize←[39;49;00m ←[39;49;00m==←[39;49;00m ←[39;49;00m←[3
4m1←[39;49;00m
←[31;01mAssertionError←[39;49;00m

Failures: ←[31;01m1←[39;49;00m/3 (0 assertions, 0.130 seconds)
```

**Comment by SimonSapin** [comment]
I’d rather do feature detection than OS detection, but I guess that’s what `isatty()` is supposed to do. After a quick search I could not find a way to test whether sys.stdout supports ANSI color codes. Anyone know if that is possible?

**Comment by eventh** [comment]
Another possible solution is to support colorama on windows, so if platform is windows, when trying to import pygments also try to import colorama, otherwise fall back on PlainReporter. But this adds another optional dependecy?

http://pypi.python.org/pypi/colorama/

**Comment by alex** [comment]
FWIW Django's color detection code looks like:

``` Python
    unsupported_platform = (sys.platform in ('win32', 'Pocket PC'))
    # isatty is not always implemented, #6223.
    is_a_tty = hasattr(sys.stdout, 'isatty') and sys.stdout.isatty()
    if unsupported_platform or not is_a_tty:
        return False
    return True
```

**Comment by eventh** [comment]
SimonSapin fixed this, so I'm closing this issue as it works now.
